services/alerts/src/useAlert.ts

```diff
@@ -1,27 +1,42 @@
-import { useContext } from 'react'
+import { useContext, useRef, useCallback } from 'react'
 import { AlertsManagerContext } from './AlertsManagerContext'
-import { Alert, AlertOptions } from './types'
+import { AlertOptions, AlertsManager, AlertsManagerAlert } from './types'
 
 export const useAlert = (
     message: string | ((props: any) => string),
     options: AlertOptions | ((props: any) => AlertOptions) = {}
 ) => {
-    const alertsManager = useContext(AlertsManagerContext)
+    const alertsManager: AlertsManager = useContext(AlertsManagerContext)
+    const alertRef = useRef(<AlertsManagerAlert>alertsManager.add())
```

Comment:
This will call `alertsManager.add()` for *every render*, even though the ref will only be initialized on mount